src/hotspot/share/gc/parallel/psOldGen.hpp

```diff
@@ -118,7 +118,7 @@ class PSOldGen : public CHeapObj<mtGC> {
   // Calculating new sizes
   void resize(size_t desired_free_space);
 
-  HeapWord* allocate(size_t word_size) {
+  HeapWord* expand_and_allocate(size_t word_size) {
     HeapWord* res;
     do {
       res = cas_allocate_noexpand(word_size);
```

Comment:
seems incorrect that we can allocate without expanding, yet the method name is "expand_and_allocate". Or you write a bit more about the motivation behind this renaming.

Reply:
`ParallelScavengeHeap::expand_heap_and_allocate` can allocate without doing expansion. The renaming is mostly for including the potential expansion in the name. Otherwise, it's not obvious that expansion is attempted in the caller context.

Reply:
So is the motivation that `ParallelScavengeHeap::expand_heap_and_allocate` already uses the name? `TenuredGeneration::expand_and_allocate` is named correctly as it expands and allocates. In cases where the expansion is optional, it appears to me that that is implementation detail that doesn't need to be part of the interface naming.